test/boost/commitlog_test.cc

```diff
@@ -650,3 +651,37 @@ SEASTAR_TEST_CASE(test_commitlog_replay_invalid_key){
         }
     });
 }
+
+using namespace std::chrono_literals;
+
+SEASTAR_TEST_CASE(test_commitlog_add_entries) {
+    return cl_test([](commitlog& log) {
+        return seastar::async([&] {
+            using force_sync = commitlog_entry_writer::force_sync;
+
+            constexpr auto n = 10;
+            for (auto fs : { force_sync(false), force_sync(true) }) {
+                std::vector<commitlog_entry_writer> writers;
+                std::vector<frozen_mutation> mutations;
+
+                writers.reserve(n);
+                mutations.reserve(n);
+                
+                for (auto i = 0; i < n; ++i) {
+                    random_mutation_generator gen(random_mutation_generator::generate_counters(false));
+                    mutations.emplace_back(gen(1).front());
+                    writers.emplace_back(gen.schema(), mutations.back(), fs);
+                }
+
+                auto res = log.add_entries(std::move(writers), db::timeout_clock::now() + 60s).get0();
+
+                std::set<segment_id_type> ids;
+                for (auto& h : res) {
+                    ids.emplace(h.rp().id);
+                }
+                BOOST_CHECK_EQUAL(ids.size(), 1);
```

Comment:
Maybe also test that we read all mutations back

Reply:
Ugh. Fine.